opencti-platform/opencti-graphql/src/migrations/1723468670664-trigger-authorized-authorities.js

```diff
@@ -0,0 +1,32 @@
+import { logApp } from '../config/conf';
+import { executionContext, SYSTEM_USER } from '../utils/access';
+import { elList, elReplace } from '../database/engine';
+import { READ_INDEX_INTERNAL_OBJECTS } from '../database/utils';
+import { ENTITY_TYPE_TRIGGER } from '../modules/notification/notification-types';
+
+const message = '[MIGRATION] update authorized authorities for triggers';
+
+export const up = async (next) => {
+  logApp.info(`${message} > started`);
+  const context = executionContext('migration');
+
+  // ------ Update triggers
+  const callback = async (notifiers) => {
+    for (let i = 0; i < notifiers.length; i += 1) {
+      const notifier = notifiers[i];
+      if (notifier.trigger_scope === 'activity') {
```

Comment:
I suggest to add this condition as a filter in `opts` (line 23), so that we directly list triggers that have this 'activity' scope, instead of listing all triggers.

Reply:
I actually think that it would be better to use `elUpdateByQueryForMigration` with a query targetting specifically triggers with scope 'activity', and updating with a script authorized_authorities. So that we update with only one query, instead of listing all triggers and updating them one by one.

Reply:
I have updated this migration with the new function `elUpdateByQueryForMigration`, and filter triggers by the trigger_scope 'activity'